### ISSUE TITLE
Change to "or" operator to improve performance and simplify code

### ISSUE THREAD

**Issue by alexfevery** [dotnet/csharplang#1885]
Many times I find myself doing something like this:

     if (DateTime.Today.DayOfWeek == DayOfWeek.Saturday || DateTime.Today.DayOfWeek == 
     DayOfWeek.Sunday) 
     {

     }

The problem with this code is it creates a new DateTime twice when only one is needed. I think it would be easy to add functionality so that I could write something like this:

    if (DateTime.Today.DayOfWeek == (DayOfWeek.Saturday || DayOfWeek.Sunday)) 
    { 

    }

Only one DateTime would be created and it would be used for both checks. runtime is almost halved and the code is also shorter.  The exact syntax could be changed to be more intuitive.

**Comment by HaloFour** [comment]
@alexfevery 

The code that you're suggesting is already legal syntax depending on the operands.  To change that behavior would be a breaking change.

There is a championed proposal to look into "or" patterns for pattern matching which will enable this kind of comparison: #1350

```cs
if (DateTime.Today.DayOfWeek is DayOfWeek.Saturday or DayOfWeek.Sunday)
{
}
```

**Comment by alexfevery** [comment]
awesome, when can we expect to see it happen?

**Comment by jnm2** [comment]
@alexfevery Another option that several folks have suggested is to write an extension method:

```c#
 if (DateTime.Today.DayOfWeek.In(DayOfWeek.Saturday, DayOfWeek.Sunday))
```

**Comment by svick** [comment]
@alexfevery https://github.com/dotnet/csharplang/issues/1350 is in the "8.X candidate" milestone. I think that means it might be included in a minor version after C# 8.0 is released. (But as I understand it, milestones are just rough plans,  there is no guarantee it will ever be included in the language.)

**Comment by gafter** [comment]
Closing as a dup of #1350.
